### ISSUE TITLE
Allow custom cron jobs

### ISSUE THREAD

**Issue by gjedeer** [nextcloud/docker#1775]
I'm looking for a way to make `occ preview:generate-all` run periodically in a Docker setup. 

Maybe adding a line like this to `/var/spool/cron/crontabs/www-data` would be a good way to enable using custom cron jobs?

```
*/5 * * * * if [ -f /var/www/html/cron_custom.sh ]; then bash /var/www/html/cron_custom.sh; fi
```

**Comment by josejsarmento** [comment]
I was looking to execute a periodic job for this exact purpose too. I was hoping for something in the README to teach us how to use the `supervisord` they embed in the `cron` Dockerfile, but I don't understand how to use it.

**Comment by gjedeer** [comment]
I had no idea there's supervisord in there. It can be turned into a poor man's cron with [a trick](https://newbedev.com/using-supervisor-as-cron) but it's very inflexible when used this way. It can be ran [every minute or every hour](http://supervisord.org/events.html#tick-60-event-type).

But I think using cron for cron jobs would be better.

**Comment by BWibo** [comment]
I was facing the same issue. For now, I'm using this workaround to run `occ preview:generate-all` every 15 mins:

The `user` is member of the docker group: `sudo usermod -a -G docker USERNAME`

* User crontag, `crontab -e`:

   ```bash
   */15 * * * * $HOME/pre-gen.sh > /dev/null 2>&1
   ```

* `$HOME/pre-gen.sh`

   ```bash
   #!/usr/bin/env bash

   echo "--- $(date --utc +%FT%TZ) ----------------------------------------" >>$HOME/pre-generate.log 2>&1
   docker exec -i --user 33 nextcloud-app-1 ./occ preview:pre-generate -v >>$HOME/pre-generate.log 2>&1
   ```

**Comment by gjedeer** [comment]
I'm doing the same but it's a hack.

**Comment by jjasoncool** [comment]
https://github.com/nextcloud/docker/issues/361

**Comment by gjedeer** [comment]
@jjasoncool yeah, building your own Dockerfile or running cronjobs on the host, not really solutions to the problem.

**Comment by jjasoncool** [comment]
https://github.com/nextcloud/docker/issues/1695#issuecomment-1383615123
